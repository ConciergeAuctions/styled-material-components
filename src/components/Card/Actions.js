// @flow
import React, { type Node } from 'react';
import styled from 'styled-components';

type CardActionsPropsType = {|
  children: Node,
  className: string,
  vertical?: boolean,
|};

const ActionsComponent = (props: CardActionsPropsType) => (
  <section className={`${props.className} card-actions`}>
    {props.children}
  </section>
);

const Actions = styled(ActionsComponent)`
  display: flex;
  box-sizing: border-box;
  padding: 8px;
  color: ${props => props.theme.textColors.primary};
  & button {
    margin: 0 8px 0 0;
  }
  ${props =>
    props.vertical &&
    `
    flex-flow: column;
    align-items: flex-start;
    & button {
      margin: 0 0 4px;
      &:last-child {
        margin-bottom: 0;
      }
    }
  `};
`;

export default Actions;
