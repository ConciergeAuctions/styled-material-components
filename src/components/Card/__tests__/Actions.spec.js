import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import {
  STYLED_COMPONENTS_CONTEXT,
  STYLED_COMPONENTS_CHILD_CONTEXT_TYPES
} from "../../../../test/utils";

import Actions from "../Actions";

describe("CardActions", () => {
  it("mounts and matches snapshot", () => {
    const mountedComponent = shallow(
      <Actions>
        <div>child 1</div>
        <div>child 2</div>
      </Actions>,
      {
        context: { ...STYLED_COMPONENTS_CONTEXT },
        childContextTypes: { ...STYLED_COMPONENTS_CHILD_CONTEXT_TYPES }
      }
    );
    expect(mountedComponent).toBeTruthy();
    expect(toJSON(mountedComponent)).toMatchSnapshot();
  });

  it("mounts and matches snapshot vertical", () => {
    const mountedComponent = shallow(
      <Actions vertical>
        <div>child 1</div>
        <div>child 2</div>
      </Actions>,
      {
        context: { ...STYLED_COMPONENTS_CONTEXT },
        childContextTypes: { ...STYLED_COMPONENTS_CHILD_CONTEXT_TYPES }
      }
    );
    expect(mountedComponent).toBeTruthy();
    expect(toJSON(mountedComponent)).toMatchSnapshot();
  });
});
