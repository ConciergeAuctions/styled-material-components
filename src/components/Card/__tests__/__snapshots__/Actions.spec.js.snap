// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardActions mounts and matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  padding: 8px;
  color: rgba(0,0,0,0.87);
}

.c0 button {
  margin: 0 8px 0 0;
}

<ActionsComponent
  className="c0"
>
  <div>
    child 1
  </div>
  <div>
    child 2
  </div>
</ActionsComponent>
`;

exports[`CardActions mounts and matches snapshot vertical 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  padding: 8px;
  color: rgba(0,0,0,0.87);
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 button {
  margin: 0 8px 0 0;
}

.c0 button {
  margin: 0 0 4px;
}

.c0 button:last-child {
  margin-bottom: 0;
}

<ActionsComponent
  className="c0"
  vertical={true}
>
  <div>
    child 1
  </div>
  <div>
    child 2
  </div>
</ActionsComponent>
`;
